src/components/AddView.js

```diff
@@ -1,91 +1,137 @@
 import React, { useState } from 'react';
 import { db } from '../lib/firebase';
 import firebase from 'firebase';
+import transformUserInput from '../lib/utils';
 
-const AddView = ({ token }) => {
-  const [item, setItem] = useState({
-    name: '',
-    howSoon: '7',
-    lastPurchasedDate: null,
-  });
+const DEFAULT_ITEM = {
+  name: '',
+  howSoon: '',
+  lastPurchasedDate: null,
+};
+
+const AddView = ({ shoppingList, token, collectionId }) => {
+  const [item, setItem] = useState(DEFAULT_ITEM);
+  const [hasDuplicates, setHasDuplicates] = useState(false);
+  const [isNameFieldEmpty, setIsNameFieldEmpty] = useState(false);
+  const [isFrequencyFieldEmpty, setIsFrequencyFieldEmpty] = useState(false);
 
   const handleChange = (e) => {
+    setHasDuplicates(false);
+    setIsNameFieldEmpty(false);
+    setIsFrequencyFieldEmpty(false);
     setItem({
       ...item,
       [e.target.name]: e.target.value,
     });
   };
 
-  const handleSubmit = (e) => {
-    e.preventDefault();
-    let docAvailable = false;
-    if (item.name.trim()) {
-      const newItem = {
-        name: item.name,
-        lastPurchasedDate: item.lastPurchasedDate,
-        howSoon: item.howSoon,
-      };
-      db.collection('shoppinglist')
-        .get()
-        .then((querySnapshot) => {
-          querySnapshot.forEach((doc) => {
-            if (doc.data().token === token) {
-              docAvailable = true;
-              doc.ref.update({
-                items: firebase.firestore.FieldValue.arrayUnion(newItem),
-              });
-            }
-          });
-          if (docAvailable === false) {
-            db.collection('shoppinglist').add({
-              items: [newItem],
-              token: token,
-            });
-          }
-        });
+  const handleAdd = async (e) => {
+    try {
+      e.preventDefault();
+      item.name = transformUserInput(item.name).toLowerCase();
 
-      setItem({
-        name: '',
-        howSoon: '7',
-        lastPurchasedDate: null,
+      const currentItems = shoppingList.map((list) => {
+        return list.items.map((item) => item.name);
       });
-    } else {
-      alert('Please write an item');
+
+      if (currentItems[0].includes(item.name)) {
+        setHasDuplicates(true);
+        return;
+      }
+
+      if (item.name === '' && item.howSoon === '') {
+        setIsNameFieldEmpty(true);
+        setIsFrequencyFieldEmpty(true);
+        return;
+      }
```

Comment:
Could this be omitted since since you check if `name` and `howSoon` are empty in the next lines?

Reply:
Hi @faysvas . We can't omit it. Without the check, when both fields are empty, an error message will be displayed for the first field only